Gruntfile.js

```diff
@@ -5,58 +5,400 @@
-// with their corresponding values, similar to C macros.
+// - Esbuild's `define` option allows to replace e.g. `process.env.VARIABLE`
+// with its corresponding value at build-time, similar to C macros.
+//
+// See https://esbuild.github.io/api/#define
 // =======================
 
-const API_PORT = 8000 + parseInt(process.env.PORT_SHIFT || 0)
-Object.assign(process.env, {
-  NODE_ENV: process.env.NODE_ENV || (process.argv.some(x => /\b(dist|deploy)\b/.test(x)) ? 'production' : 'development'),
-  API_PORT,
-  // TODO: make the protocol (http vs https) variable based on environment var
-  API_URL: 'http://localhost:' + API_PORT
-})
-
-// =======================
-// Trampoline to enable babel in various places
-// (the backend, mocha tests, .Gruntfile.babel.js)
-// =======================
+/**
+ * Creates a modified copy of the given `process.env` object, according to its `PORT_SHIFT` variable.
+ *
+ * The `API_PORT` and `API_URL` variables will be updated.
+ * TODO: make the protocol (http vs https) variable based on environment var.
+ * @param {Object} env
+ * @returns {Object}
+ */
+const applyPortShift = (env) => {
+  // TODO: implement automatic port selection when `PORT_SHIFT` is 'auto'.
+  const API_PORT = 8000 + Number.parseInt(env.PORT_SHIFT || '0')
+  const API_URL = 'http://localhost:' + API_PORT
 
-// Now 'require' is primed to parse files through babel.
-if (!process.env.LOAD_NO_FILE) {
-  const target = process.env.LOAD_TARGET_FILE || './.Gruntfile.babel.js'
-  console.log('Gruntfile.js: loading file with babel support:', target)
-  module.exports = require(target)
-} else {
-  console.log('Gruntfile.js: just calling babel-register, not loading any file.')
+  if (Number.isNaN(API_PORT) || API_PORT < 8000 || API_PORT > 65535) {
+    throw new RangeError(`Invalid API_PORT value: ${API_PORT}.`)
+  }
+  return { ...env, API_PORT, API_URL }
 }
 
-// =======================
-// Old comments, that used to exist at the top of this file,
-// are saved here for "safe keeping". These might be outdated
-// and certainly will be removed at somepoint.
-// =======================
+Object.assign(process.env, applyPortShift(process.env))
+
+const chalk = require('chalk')
+const crypto = require('crypto')
+const { fork } = require('child_process')
+const fs = require('fs')
+const path = require('path')
+const { resolve } = path
+
+// Not loading babel-register here since it is quite a heavy import and is not always used.
+// We will rather load it later, and only if necessary.
+// require('@babel/register')
+
+const { NODE_ENV = 'development', VUEX_STRICT = 'true' } = process.env
+
+const backendIndex = './backend/index.js'
+const distAssets = 'dist/assets'
+const distCSS = 'dist/assets/css'
+const distDir = 'dist'
+const distJS = 'dist/assets/js'
+// const serviceWorkerDir = 'frontend/controller/serviceworkers'
+const srcDir = 'frontend'
+
+const chompLeft = (s, what) => s.startsWith(what) ? s.slice(what.length) : s
```

Comment:
Why is this being rewritten instead of imported as before?

Reply:
Importing wouldn't work here as babel is not loaded yet at this point.

Reply:
Anyway, `chompLeft()` is only used in the Sass plugin `importer` option, which has no effect for some reason (it looks like the Vue plugin takes precedence over the Sass one except for the `main.scss` file), so we might remove it later.

Reply:
ah I see. Can you add a comment to that effect above it?